### PR TITLE
LPS-188855 Create model resource for ObjectModel

### PR DESCRIPTION
**Pull request by murilostodolni**
Related: https://liferay.atlassian.net/browse/LPS-188855

We will need solutions team review in this https://github.com/liferay-objects/liferay-portal/pull/2299/commits/caec96bf8be806607f9ffd73f8e95d5e18d0d6fa commit and QA BPM review in these https://github.com/liferay-objects/liferay-portal/pull/2299/commits/9557ec72f59e05ca1f9c65dc12fd8d192fba3c33  https://github.com/liferay-objects/liferay-portal/pull/2299/commits/3673e1eee5e6544131281af2528889323dbd72bb commits.

### CODE REVIEW

File: modules/apps/object/object-api/src/main/java/com/liferay/object/constants/ObjectActionKeys.java [hunk 1]
```diff
@@ -26,6 +26,8 @@ public class ObjectActionKeys {
 	public static final String EXTEND_SYSTEM_OBJECT_DEFINITION =
 		"EXTEND_SYSTEM_OBJECT_DEFINITION";
 
+	public static final String MANAGE_MODELS = "MANAGE_MODELS";
```
Comment (seltong) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/constants/ObjectActionKeys.java:29]: Hey, I'm a little bit apprehensive about the name of this constant. What do you think about the options below?

- MODELS_MANAGER
- MODELS_MANAGEMENT

If you have any other options, feel free to share then so we can decide together the best name for this constant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/deployer/ObjectDefinitionDeployerImpl.java [hunk 2]
```diff
@@ -243,7 +243,7 @@ public List<ServiceRegistration<?>> deploy(
 					_resourcePermissionLocalService,
 					_userGroupRoleLocalService),
 				HashMapDictionaryBuilder.<String, Object>put(
-					"com.liferay.object", "true"
+					"com.liferay.object.definition", "true"
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/deployer/ObjectDefinitionDeployerImpl.java:246]: Could you please check if we can use here the `ObjectConstants.RESOURCE_NAME_OBJECT_DEFINITION`?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/deployer/ObjectDefinitionDeployerImpl.java [hunk 3]
```diff
@@ -297,7 +297,7 @@ public List<ServiceRegistration<?>> deploy(
 			_bundleContext.registerService(
 				PortletResourcePermission.class, portletResourcePermission,
 				HashMapDictionaryBuilder.<String, Object>put(
-					"com.liferay.object", "true"
+					"com.liferay.object.definition", "true"
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/deployer/ObjectDefinitionDeployerImpl.java:300]: Could you please check if we can use here the `ObjectConstants.RESOURCE_NAME_OBJECT_DEFINITION`?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java [hunk 4]
```diff
@@ -408,7 +408,7 @@ protected void activate(
 		_portletResourcePermissionsServiceTrackerMap =
 			ServiceTrackerMapFactory.openSingleValueMap(
 				bundleContext, PortletResourcePermission.class,
-				"(&(com.liferay.object=true)(resource.name=*))",
+				"(&(com.liferay.object.definition=true)(resource.name=*))",
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java:411]: Could you please check if we can use here the `ObjectConstants.RESOURCE_NAME_OBJECT_DEFINITION`?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v5_4_0/ResourcePermissionUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+					"select objectDefinitionId from ObjectDefinition where " +
+						"system_ = [$FALSE$]"));
+			ResultSet resultSet = preparedStatement.executeQuery()) {
+
+			while (resultSet.next()) {
+				long objectDefinitionId = resultSet.getLong(
+					"objectDefinitionId");
+
+				String newName =
+					ObjectConstants.RESOURCE_NAME_OBJECT_DEFINITION +
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v5_4_0/ResourcePermissionUpgradeProcess.java:44]: Use StringBundler here

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v5_4_0/ResourcePermissionUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+			}
+		}
+
+		updateResourceAction(
+			"com.liferay.object",
+			ObjectConstants.RESOURCE_NAME_OBJECT_DEFINITION);
+
+		updateResourcePermission(
+			"com.liferay.object",
+			ObjectConstants.RESOURCE_NAME_OBJECT_DEFINITION, true);
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v5_4_0/ResourcePermissionUpgradeProcess.java:60]: We are repeating the "com.liferay.object" in the code. How about creating a `private static final String _RESOURCE_NAME = "com.liferay.object"` in this class and using it on lines 46, 55 and 59?

**Review by seltong — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-api/src/main/java/com/liferay/object/constants/ObjectActionKeys.java (hunk 1 above)
Comment (murilostodolni) [reply modules/apps/object/object-api/src/main/java/com/liferay/object/constants/ObjectActionKeys.java:29]: Hey Selton, `MANAGE_MODELS` was defined in mock-up.

**Comment by murilostodolni** [comment]
ci:test:sf

**Review by seltong — APPROVED** [review]
LGTM :tada:

**Comment by murilostodolni** [comment]
ci:test:object

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:close

**Pull request closed without merging**